### ISSUE TITLE
[Feature] Allow multiple weapons nodes on a single outfit

### ISSUE THREAD

**Issue by Azure3141** [endless-sky/endless-sky#7408]
### Problem Description

Outfits can only have a single `weapon` node, meaning to have multiple effects (e.g. anti-ship damage, antimissile, tractor beams) multiple separate outfits and hardpoints are needed. While not inherently a problem with the game, it would open up additional possibilities if we could add multiple `weapon` nodes to the same outfit.

To be specific, I plan on allowing the Avgi Blue and Green Optical Lasers work as both antimissile turrets and anti-ship turrets (balancing this will be fun). I also would like to add the ability for the same missile launcher, e.g. the Vertical Launch Array, to be loaded with combinations of other (new) missiles.

### Related ~~Issue~~ PR Links

Probably should come after #9859 cleans things up.

### Desired Solution

Allow a single outfit to have multiple `weapon` nodes defined without them overwriting each other. Save them to a list tied to the outfit instead, with both being placed at the same hardpoint when installed on a ship.

Some kind of prioritization keyword would also be useful. For example, if an Optical Laser is shooting at a missile, and you hit the fire key to shoot at an enemy ship, the turret would stop firing at the missile and prioritize the anti-ship function instead. Alternatively, you could leave the prioritization attribute off and make an antimissile system with multiple identical `weapon` nodes capable of targeting multiple missiles at the same time.

### Alternative Approaches

N/A

### Additional Context

Its woulds bees muchs cools

**Comment by ThrawnCA** [comment]
Isn't this handled by just having multiple turret slots?

**Comment by Zitchas** [comment]
Not really. There is a fairly significant difference between AAA that can shoot at missiles and fighters, and having an anti missile and a fast turret sitting next to each other.

It would be nice to support multiple weapon outfits, anyway.

**Comment by beccabunny** [comment]
How would turret turning work? AM rotated instantly while normal turrets rotate with a specific rate, which means that the two would conflict if it were to switch modes quickly

**Comment by ThrawnCA** [comment]
> There is a fairly significant difference between AAA that can shoot at missiles and fighters, and having an anti missile and a fast turret sitting next to each other.

Well, yes, a turret that can do both, but only takes up one slot, would certainly be more efficient/effective. But I would have said that the turret limitation is intentional. If a ship only has one turret slot available, then it's not supposed to be advanced enough to support multiple turret functions.

**Comment by vitalchip** [comment]
Mixed feelings on this one, certainly turrets which perform both ship damage and AM are useful, if for no other reason than I no longer need to fly around in a lop sided Blackbird.

Large ships are woefully short of turrets and that should be the answer for them.

Smaller ships seem like a more reasonable case, combination turrets are always going to be a compromise anyway, due to engine limitations they're only viable for beam weapons because AM is beam only. So if it's meant to be the same weapon being double purposed then it will share most characteristics, including reload. So you potentially face having no shot available against another ship because the weapon is still reloading from shooting at a missile.

If it's meant to be two co-located weapons (individual weapon type, reoad, range etc) then just co-located two turrets, we need nothing new.

**Comment by Zitchas** [comment]
Well, there is no reason that had to be the case... There could be multiple barrels mounted on the platform and it picks the appropriate one based on target. Good design, maybe not, but still possible.

Although this really makes me miss EVN's mechanics, where they didn't have anti-missile, they had Anti Air Artillery. As in things like quad light blaster turrets, which functioned pretty much exactly like the CWS and naturally would fire clouds of shots at missiles and fighters alike. But then EVN's missiles actually had hp of a sort. Interestingly, the AAA weapons main restriction was that they ignored anything bigger than a fighter.

**Comment by ixPost** [comment]
I think this would be good for Quarg antimissile.
